tests/cli.rs

```diff
@@ -0,0 +1,167 @@
+use vrf_dalek::golden::{GoldenTestVector, CARDANO_BASE_TEST_VECTORS};
+
+use assert_cmd::Command;
+use predicates::prelude::*;
+use std::fs;
+
+const PRG: &str = "vrf_dalek";
+
+#[test]
+fn correct_output_help_arg() {
+    let mut cmd = Command::cargo_bin(PRG).unwrap();
+    cmd.arg("--help")
+        .assert()
+        .success()
+        .stdout(predicate::str::contains("USAGE"));
+}
+
+#[test]
+fn correct_output_version_arg() {
+    let mut cmd = Command::cargo_bin(PRG).unwrap();
+    let ver = "vrf_dalek 0.1.0";
+    cmd.arg("--version")
+        .assert()
+        .success()
+        .stdout(predicate::str::contains(ver));
+}
+
+#[test]
+fn correct_length_hex_output_generate_arg() {
+    let mut cmd = Command::cargo_bin(PRG).unwrap();
+    let is_32_byte_hex = predicate::str::is_match("^[0-9a-f]{64}$").unwrap();
+    cmd.arg("--generate")
+        .assert()
+        .success()
+        .stdout(is_32_byte_hex);
+}
+
+#[test]
+fn check_publickey_deriving_from_file_with_golden_tests() {
+    for filename in CARDANO_BASE_TEST_VECTORS {
+        let _ = check_deriving_against_golden_from_file(&filename);
+    }
+}
+
+#[test]
+fn check_publickey_deriving_from_stdin_with_golden_tests() {
+    for filename in CARDANO_BASE_TEST_VECTORS {
+        let _ = check_deriving_against_golden_from_stdin(&filename);
+    }
+}
+
+#[test]
+fn check_proving_with_golden_tests() {
+    for filename in CARDANO_BASE_TEST_VECTORS {
+        let _ = check_proving_against_golden(&filename);
+    }
+}
+
+#[test]
+fn check_hash_output_from_file_with_golden_tests() {
+    for filename in CARDANO_BASE_TEST_VECTORS {
+        let _ = check_hash_output_against_golden_from_file(&filename);
+    }
+}
+
+#[test]
+fn check_hash_output_from_stdin_with_golden_tests() {
+    for filename in CARDANO_BASE_TEST_VECTORS {
+        let _ = check_hash_output_against_golden_from_stdin(&filename);
+    }
+}
+
+#[test]
+fn check_verifying_with_golden_tests() {
+    for filename in CARDANO_BASE_TEST_VECTORS {
+        let _ = check_verifying_against_golden(&filename);
+    }
+}
+
+fn check_deriving_against_golden_from_file(file_path: &str) {
+    let mut cmd = Command::cargo_bin(PRG).unwrap();
+    let input = fs::read_to_string(file_path).unwrap();
+    let golden = serde_json::from_str::<GoldenTestVector>(&input).unwrap();
+    fs::write("sk1.prv", &hex::encode(&golden.secret_key)).unwrap();
```

Comment:
suggestion: use something like [tempfile](https://docs.rs/tempfile/latest/tempfile/fn.tempfile.html) to avoid cluttering the current directory?

Reply:
good suggestion - thanks. done!